dali/pipeline/operator/sequence_operator.h

```diff
@@ -36,8 +37,62 @@ inline bool is_per_frame(const TensorVector<CPUBackend> &arg_input) {
   const auto &layout = arg_input.GetLayout();
   return layout.size() > 0 && layout[0] == 'F';
 }
+
+template <typename BatchType>
+struct BatchBackend;
+
+template <template <typename> class BatchContainer, typename Backend>
+struct BatchBackend<BatchContainer<Backend>> {
+  using Type = Backend;
+};
+
+template <typename BatchType>
+using batch_backend_t =
+    typename BatchBackend<std::remove_cv_t<std::remove_reference_t<BatchType>>>::Type;
```

Comment:
Does a particular suitable file come to your head?

Reply:
Actually, I have mixed feelings, because it can backfire terribly. Effectively, it's just a utility that extracts the type template argument from a single-type template. Just see:
```C++
    std::mutex m;
    std::lock_guard<std::mutex> g(m);
    static_assert(std::is_same<batch_backend_t<decltype(g)>, std::mutex>::value);
```
If we were to make it a general utility, we could make `Backend` a `typedef` inside classes templated against Backend (Buffer, TensorXxx, Operator) and then this trait could be defined as
```C++
template <typename X>
using backend_t = typename X::Backend;
```
or possibly
```C++
template <typename X>
using backend_t = typename std::remove_cv_t<std::remove_reference_t<X>>::Backend;
```

Reply:
Having said all that, I'm fine with keeping it as is in this limited scope - perhaps this utility should even be moved inside the class, so it's not (ab)used elsewhere.

Reply:
Okay, so I moved it to the type_traits.h and added enable_if_t<is_batch_container<...>> guard, this way it can be reused and to abused.